airflow/providers/amazon/aws/sensors/dynamodb.py

```
@@ -0,0 +1,85 @@
+    :param sort_key_name: (optional) DynamoDB sort key name
+    :param sort_key_value: (optional) DynamoDB sort key value
+    """
+
+    def __init__(
+        self,
+        table_name: str,
+        partition_key_name: str,
+        partition_key_value: str,
+        attribute_name: str,
+        attribute_value: str,
+        sort_key_name: Optional[str] = None,
+        sort_key_value: Optional[str] = None,
+        aws_conn_id: str = "aws_default",
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.table_name = table_name
+        self.partition_key_name = partition_key_name
+        self.partition_key_value = partition_key_value
+        self.attribute_name = attribute_name
+        self.attribute_value = attribute_value
+        self.sort_key_name = sort_key_name
+        self.sort_key_value = sort_key_value
+        self.aws_conn_id = aws_conn_id
+        self.hook = DynamoDBHook(aws_conn_id=self.aws_conn_id)
+        self.table = self.hook.get_conn().Table(table_name)
+
+    def poke(self, context: Context) -> bool:
+        """Test DynamoDB item for matching attribute value"""
+        try:
+            key = {self.partition_key_name: self.partition_key_value}
+            msg = f"Checking table {self.table_name} for item PK: {self.partition_key_name}={self.partition_key_value}"
+
+            if self.sort_key_value:
+                key[self.sort_key_name] = self.sort_key_value
+                msg += f" SK: {self.sort_key_name}={self.sort_key_value}"
+
+            msg += f" attribute: {self.attribute_name}={self.attribute_value}"
+
+            self.log.info(msg)
+            response = self.table.get_item(Key=key)
+            self.log.info(f"Response: {response}")
+            return response["Item"][self.attribute_name] == self.attribute_value
+        except ClientError as ex:
+            self.log.error('AWS request failed, check logs for more info: %s', ex)
+            return False
```

Do we want to keep looping if we're getting an exception back from the api?

---

I agree, we need to catch only specific errors. Because a lot of `ClientError` shouldn't retried. For example if user provide wrong credentials:

```console
'ClientError' error occurred while testing connection: An error occurred (IncompleteSignature)
```
```console
'ClientError' error occurred while testing connection: An error occurred (InvalidClientTokenId)
```

---

Are you suggesting it's best to not catch an exception at all here? Looking at the other sensors' code, it doesn't look like those catch exceptions.

---

> Are you suggesting it's best to not catch an exception at all here?

Moer that we should only be catching and handling (in this case ignoring) specific/expected exceptions rather than any possible ClientError. Like Andrey listed above, there are many exceptions that if received do not make sense to retry.

So if there is no condition that you specifically have in mind here, I would just drop the exception handling and let it bubble up to the user to notify them that something failed.

---

That's exactly what I did in this case — I dropped the exception handling altogether.